### PR TITLE
Feature: Rclone support

### PR DESCRIPTION
**Pull request by gustavocardosomoreira**
Note: This PR depends on #6 , #7, #8 and #9, please review them first.

## Description

This feature implements rclone support. 
Set the following keys to use it:
- "remote_sync_service=rclone"
- "rclone_remote" and "rclone_path": Use the same name as used in rclone's rclone.conf file and the path in that remote.
- "rclone_someflag": Use any environment variable as described [here](https://rclone.org/docs/#environment-variables) and [here](https://rclone.org/flags/). Use lowercase keys, the software will automatically convert it to uppercase.

## Motivation and Context

I needed to sync to the cloud. Rclone is one of the best options as it supports hundreds of cloud providers.

## How has this been tested?

Main, test and test_config pass in my system python and in Travis-CI.
Haven't tested a full torrent workflow (add, download and remotesync)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: remotesync.py [hunk 1]
```diff
@@ -166,3 +171,123 @@ def rsync_command(self):
...
+        self.RCLONE_REMOTE = kwargs.pop('rclone_remote')
+        self.RCLONE_PATH = kwargs.pop('rclone_path')
+
+        # Rclone config flags can be set entirely using environment variables.
+        # This is what is done in this implementation.
+        # https://rclone.org/docs/#environment-variables
+        # Warn user that he is overriding some default flags.
+        flags = set(self.DEFAULT_FLAGS.keys()) & set(kwargs.keys())
+        if flags:
+            warning('Flags %s were passed, these might mess up log formatting, thread carefully!' % flags)
```
Comment (amoe) [inline remotesync.py:195]: Typo, I guess it should be 'tread carefully' :)

**Review by amoe — CHANGES_REQUESTED** [review]

File: remotesync.py (hunk 1 above)
Comment (gustavocardosomoreira) [reply remotesync.py]: Eagle eyes, Mr Banks. :smile: Fixed.

File: remotesync.py [hunk 2]
```diff
@@ -166,3 +171,123 @@ def rsync_command(self):
...
+
+        # Store all flags that apply to rclone.
+        # User-defined flags replace default ones.
+        self.rclone_flags = {
+            **self.DEFAULT_FLAGS,
+            **{k.upper(): v for k, v in kwargs if k.startswith('rclone_')}
```
[the remark this answers is not in the document]
Comment (gustavocardosomoreira) [reply remotesync.py]: Fixed. The error comes up if one runs the software with `remote_sync_interface=rclone`, I had tested it many times before with `rclone`, but at the time of this commit was running the software with a config file with `remote_sync_interface=rsync` and didn't test again it with both keys. 
This was a mistake on my part for developing a feature without implementing some automated testing to cover that code. Sorry. 
But, I think this commit is already too big to implement said testing here.
Cheers Mr Banks.

**Review by amoe — APPROVED** [review]
Tested, looks good :tada:

**Comment by amoe** [comment]
Just wanted to say, thanks again for this PR -- it's a really useful feature, and it must have been a lot of work.

**Pull request merged**